Gordon360/Services/NewsService.cs

```diff
@@ -240,27 +239,50 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
 
             if (newData.Image != null)
             {
-                string fileName = newsItem.SNID + ".jpg";
-                string imagePath = NewsUploadsPath + fileName;
-                ImageUtils.UploadImage(imagePath, newData.Image);
-                newsItem.Image = imagePath;
+                // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
+                var (extension, format, data) = ImageUtils.GetImageFormat(newData.Image);
+
+                string? imagePath = null;
+                // If old image exists, overwrite it with new image at same path
+                if (newsItem.Image != null)
+                {
+                    imagePath = GetImagePath(Path.GetFileName(newsItem.Image));
+                }
+                // Otherwise, upload new image and save url to db
+                else
+                {
+                    // Use a unique alphanumeric GUID string as the file name
+                    var filename = $"{Guid.NewGuid().ToString("N")}.{extension}";
+                    imagePath = GetImagePath(filename);
+                    var serverAddress = _serverUtils.GetAddress();
+                    if (serverAddress is not string) throw new Exception("Could not upload Student News Image: Server Address is null");
+                    var url = $"{serverAddress}/browseable/uploads/{filename}";
+                    newsItem.Image = url;
+                }
+
+                ImageUtils.UploadImage(imagePath, data, format);
             }
 
             //If the image property is null, it means that either the user
             //chose to remove the previous image or that there was no previous
             //image (DeleteImage is designed to handle this).
-            else
+            else if (newsItem.Image != null)
             {
-                string fileName = newsItem.SNID + ".jpg";
-                string imagePath = NewsUploadsPath + fileName;
+                var imagePath = GetImagePath(Path.GetFileName(newsItem.Image));
+
                 ImageUtils.DeleteImage(imagePath);
-                newsItem.Image = newData.Image;//null
+                newsItem.Image = newData.Image; //null
             }
             _context.SaveChanges();
 
             return newsItem;
         }
 
+        public string GetImagePath(string filename)
+        {
+            return Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", filename);
```

Comment:
One last thing: we should upload news images to the `browseable/uploads/news` folder, to distinguish them from other uploads.
```suggestion
            return Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", "news", filename);
```